trinity/extensibility/plugin_manager.py

```diff
@@ -74,7 +113,55 @@ def broadcast(self, event: BaseEvent, exclude: BasePlugin = None) -> None:
             if not plugin.should_start():
                 continue
 
-            plugin.start(None)
+            plugin._start()
             self._started_plugins.append(plugin)
             self._logger.info("Plugin started: {}".format(plugin.name))
             self.broadcast(PluginStartedEvent(plugin), plugin)
+
+    def prepare(self,
+                args: Namespace,
+                chain_config: ChainConfig,
+                boot_kwargs: Dict[str, Any] = None) -> None:
+        for plugin in self._plugin_store:
+
+            if not self._is_responsible_for_plugin(plugin):
+                continue
+
+            context = self._create_context_for_plugin(plugin)
+            context.args = args
+            context.boot_kwargs = boot_kwargs
+            context.chain_config = chain_config
+            plugin.set_context(context)
+
+    def shutdown(self) -> None:
+        for plugin in self._started_plugins:
+            plugin.stop()
```

Comment:
Maybe wrap this in a `try/except` so that if a single plugin throws it doesn't prevent a clean shutdown on the remaining plugins?

Reply:
Seems right. And then just issue a warning, right?